server/mailet/rate-limiter-redis/src/main/java/org/apache/james/rate/limiter/redis/RedisRateLimiterConfiguration.java

```diff
@@ -0,0 +1,106 @@
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.rate.limiter.redis;
+
+import java.time.Duration;
+
+import org.apache.commons.configuration2.HierarchicalConfiguration;
+import org.apache.commons.configuration2.tree.ImmutableNode;
+
+import com.google.common.base.Preconditions;
+
+import io.lettuce.core.RedisURI;
+
+public class RedisRateLimiterConfiguration {
+    public static RedisRateLimiterConfiguration parse(HierarchicalConfiguration<ImmutableNode> configuration) {
+        return builder()
+            .redisURI(configuration.getString("redisURL", null))
+            .periodTime(configuration.getLong("period", null))
+            .redisKeyPrefix(configuration.getString("redisKeyPrefix", null))
+            .build();
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private RedisURI redisURI;
+        private String redisKeyPrefix;
+        private Duration periodTime;
+
+        public Builder() {
+        }
+
+        public Builder redisURI(RedisURI redisURI) {
+            this.redisURI = redisURI;
+            return this;
+        }
+
+        public Builder redisURI(String redisURI) {
+            return redisURI(RedisURI.create(redisURI));
+        }
+
+        public Builder redisKeyPrefix(String redisKeyPrefix) {
+            this.redisKeyPrefix = redisKeyPrefix;
+            return this;
+        }
+
+        public Builder periodTime(Duration periodTime) {
+            this.periodTime = periodTime;
+            return this;
+        }
+
+        public Builder periodTime(long periodTime) {
+            return periodTime(Duration.ofSeconds(periodTime));
+        }
+
+        public RedisRateLimiterConfiguration build() {
+            return new RedisRateLimiterConfiguration(redisURI, redisKeyPrefix, periodTime);
+        }
+
+    }
+
+    public RedisRateLimiterConfiguration(RedisURI redisURI, String redisKeyPrefix, Duration periodTime) {
+        Preconditions.checkNotNull(redisURI);
+        Preconditions.checkNotNull(redisKeyPrefix);
+        Preconditions.checkNotNull(periodTime);
+        Preconditions.checkArgument(!periodTime.isNegative() && !periodTime.isZero(), "period must be a positive");
+        this.redisURI = redisURI;
+        this.redisKeyPrefix = redisKeyPrefix;
+        this.periodTime = periodTime;
+    }
+
+    private final RedisURI redisURI;
+    private final String redisKeyPrefix;
+    private final Duration periodTime;
+
+    public RedisURI getRedisURI() {
+        return redisURI;
+    }
+
+    public String getRedisKeyPrefix() {
+        return redisKeyPrefix;
+    }
+
+    public Duration getPeriodTime() {
+        return periodTime;
+    }
```

Comment:
Not used, correct?

Reply:
Maybe, we have `org.apache.james.rate.limiter.api.Rule(quantity: Quantity, duration: Duration)`

Reply:
I mean this parameter comes from the usage not from the Redis config as far as I understand.